# Library Book Recommendation System

In [1]:
#Importing the necessary libraries
import pandas as pd
import numpy as np

In [2]:
#Loading the datasets
rt_df = pd.read_csv("data/BX-Book-Ratings.csv", sep=";", error_bad_lines=False, encoding="latin-1")
bk_df = pd.read_csv("data/BX-Books.csv", sep=";", error_bad_lines=False, encoding="latin-1")
us_df = pd.read_csv("data/BX-Users.csv", sep=";", error_bad_lines=False, encoding="latin-1")

/var/folders/nn/vxj31dzx2ws_fw98wqh_r6wh0000gp/T/ipykernel_14081/1103073950.py:1: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.


  rt_df = pd.read_csv("data/BX-Book-Ratings.csv", sep=";", error_bad_lines=False, encoding="latin-1")
/var/folders/nn/vxj31dzx2ws_fw98wqh_r6wh0000gp/T/ipykernel_14081/1103073950.py:2: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.


  bk_df = pd.read_csv("data/BX-Books.csv", sep=";", error_bad_lines=False, encoding="latin-1")
b'Skipping line 6452: expected 8 fields, saw 9\nSkipping line 43667: expected 8 fields, saw 10\nSkipping line 51751: expected 8 fields, saw 9\n'
b'Skipping line 92038: expected 8 fields, saw 9\nSkipping line 104319: expected 8 fields, saw 9\nSkipping line 121768: expected 8 fields, saw 9\n'
b'Skipping line 144058: expected 8 fields, saw 9\nSkipping line 15078

In [3]:
rt_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1149780 entries, 0 to 1149779
Data columns (total 3 columns):
 #   Column       Non-Null Count    Dtype 
---  ------       --------------    ----- 
 0   User-ID      1149780 non-null  int64 
 1   ISBN         1149780 non-null  object
 2   Book-Rating  1149780 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 26.3+ MB


In [4]:
rt_df.isna().sum()

User-ID        0
ISBN           0
Book-Rating    0
dtype: int64

In [8]:
rt_df.duplicated().sum()

0

**Ratings Data**:
>* Has **3 columns** and **1,149,780** rows.
>* Has 2 datatypes : **int64** and **object**
>* Has no missing values
>* Has no duplicates

In [11]:
bk_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271360 entries, 0 to 271359
Data columns (total 8 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   ISBN                 271360 non-null  object
 1   Book-Title           271360 non-null  object
 2   Book-Author          271359 non-null  object
 3   Year-Of-Publication  271360 non-null  object
 4   Publisher            271358 non-null  object
 5   Image-URL-S          271360 non-null  object
 6   Image-URL-M          271360 non-null  object
 7   Image-URL-L          271357 non-null  object
dtypes: object(8)
memory usage: 16.6+ MB


In [15]:
bk_df.isna().sum()

ISBN                   0
Book-Title             0
Book-Author            1
Year-Of-Publication    0
Publisher              2
Image-URL-S            0
Image-URL-M            0
Image-URL-L            3
dtype: int64

In [12]:
bk_df.duplicated().sum()

0

**Books Data**:
>* Has **8 columns** and **271,359** rows.
>* Has 1 datatype: **object**
>* Has missing values:
 >>* Book author: 1 
 >>* Publisher: 2
 >>* Image-URL-L: 3
>* Has no duplicates

In [16]:
us_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 278858 entries, 0 to 278857
Data columns (total 3 columns):
 #   Column    Non-Null Count   Dtype  
---  ------    --------------   -----  
 0   User-ID   278858 non-null  int64  
 1   Location  278858 non-null  object 
 2   Age       168096 non-null  float64
dtypes: float64(1), int64(1), object(1)
memory usage: 6.4+ MB


In [17]:
us_df.isna().sum()

User-ID          0
Location         0
Age         110762
dtype: int64

In [14]:
us_df.duplicated().sum()

0

**Users Data**:
>* Has **3 columns** and **278,857** rows.
>* Has 3 datatypes **int64**, **float64** and **object**
>* Has missing values:
 >>* Age: 110,762
>* Has no duplicates